### PR TITLE
Updating dints config to distinguish training and searching hyper-parameters

### PR DESCRIPTION
**Pull request by dongyang0122**
Signed-off-by: Dong Yang <dongy@dongy-mlt.client.nvidia.com>

### CODE REVIEW

File: auto3dseg/algorithm_templates/dints/scripts/train.py [hunk 1]
```diff
@@ -46,21 +46,21 @@ def run(config_file: Optional[Union[str, Sequence[str]]] = None, **override):
...
-    fold = parser.get_parsed_content("fold")
-    num_images_per_batch = parser.get_parsed_content("num_images_per_batch")
-    num_iterations = parser.get_parsed_content("num_iterations")
-    num_iterations_per_validation = parser.get_parsed_content("num_iterations_per_validation")
-    num_sw_batch_size = parser.get_parsed_content("num_sw_batch_size")
-    output_classes = parser.get_parsed_content("output_classes")
-    overlap_ratio = parser.get_parsed_content("overlap_ratio")
-    patch_size_valid = parser.get_parsed_content("patch_size_valid")
-    softmax = parser.get_parsed_content("softmax")
+    amp = parser.get_parsed_content("training#trainingamp")
```
Comment (mingxin-zheng) [inline auto3dseg/algorithm_templates/dints/scripts/train.py:49]: Is it `training@amp`?

File: auto3dseg/algorithm_templates/dints/configs/hyper_parameters.yaml [hunk 2]
```diff
@@ -1,61 +1,61 @@
...
+    _target_: torch.optim.lr_scheduler.StepLR
+    optimizer: "$@training#optimizer"
+    step_size: "$@training#num_iterations // 5"
+    gamma: 0.5
 
 # fine-tuning
 finetune:
   activate: false
-  pretrained_ckpt_name: "$@bundle_root + '/model_fold' + str(@fold) + '/best_metric_model.pt'"
+  pretrained_ckpt_name: "$@training#bundle_root + '/model_fold' + str(@training#fold) + '/best_metric_model.pt'"
```
Comment (mingxin-zheng) [inline auto3dseg/algorithm_templates/dints/configs/hyper_parameters.yaml:48]: Just to confirm, the folder structure will be something like:
```bash
└── dints0
    ├── arch_ram_fold0
    ├── configs
    ├── model_fold0
    └── scripts
└── dints1
    ├── arch_ram_fold1
    ├── configs
    ├── model_fold1
    └── scripts
```
It is fine by me

File: auto3dseg/algorithm_templates/dints/scripts/train.py (hunk 1 above)
Comment (dongyang0122) [reply auto3dseg/algorithm_templates/dints/scripts/train.py]: corrected. thanks!

File: auto3dseg/algorithm_templates/dints/configs/hyper_parameters.yaml (hunk 2 above)
Comment (dongyang0122) [reply auto3dseg/algorithm_templates/dints/configs/hyper_parameters.yaml:48]: yes. it is expected.

Comment (mingxin-zheng) [reply auto3dseg/algorithm_templates/dints/configs/hyper_parameters.yaml:48]: Can we one bundle_root for training and searching, if it is expected to share the same folder?

Comment (mingxin-zheng) [reply auto3dseg/algorithm_templates/dints/configs/hyper_parameters.yaml:48]: Another benefit of keeping one bundle_root for both training and searching: the user can use the following syntax for dints and other algorithm to run the bundle

```bash
torchrun --nnodes=1 \
    --nproc_per_node=${N_GPU} -m \
    scripts.train run \
    --config_file ...
    --bundle_root ${FOLDER_PATH} "
```

Comment (dongyang0122) [reply auto3dseg/algorithm_templates/dints/configs/hyper_parameters.yaml:48]: that's fair.

Comment (dongyang0122) [reply auto3dseg/algorithm_templates/dints/configs/hyper_parameters.yaml]: just update it.

**Review by mingxin-zheng — APPROVED** [review]

**Pull request merged**